### DISCUSSION TITLE
CommandBase requires [Create] method?

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#2297] [Questions]
The following used to be allowed:

```cs
var cmd = new MyCommand();

cmd     = DataPortal.Execute(cmd);

return cmd.Result;
```

This now produces a `CSLA0011: CSLA business objects should not be created outside of a ObjectFactory instance` error, so I assume the fix is:

```cs
var cmd = DataPortal.Create<MyCommand>();
```

However, this also throws without an empty `[Create]` method, but there's no warning or error from the CSLA analyzers.
```cs
[Create]
[RunLocal]
private void Create()
{
}
```

Is this the only solution? Is there any reason why I shouldn't inherit `CommandBase` to add a virtual `Create()` method?

**EDIT**
The `[Create]` requirement seems a bit strange given that `CommandBase` treats `DataPortal_Create` as unsupported:
```cs
private void DataPortal_Create(object criteria)
{
    throw new NotSupportedException(Resources.CreateNotSupportedException);
}
```

**Comment by rockfordlhotka** [comment]
In CSLA 6 the base class methods are being entirely removed, as they are superfluous.

You should be able to implement an empty `[Create]` method in your base class, because in CSLA 5 operation attributes have priority over the legacy DataPortal_XYZ methods.

**Reply by michaelcsikos** [reply]
For a `CommandBase` it might be nice to allow a `DataPortal.Create<T>` without a `[Create]` method, which would be the equivalent of `new` with the default constructor in older versions of CSLA.
